dev/com.ibm.ws.st.core/src/com/ibm/ws/st/core/internal/looseconfig/LooseArchive.java

```diff
@@ -336,27 +321,9 @@ else if (isMatchedComponentDependency(file, project.getLocation(), javaClasspath
 
     }
 
-    private boolean isMatchedComponentDependency(File file, IPath projPath, IClasspathEntry[] javaClasspathEntries) {
-        for (IClasspathEntry entry : javaClasspathEntries) {
-            final IClasspathAttribute[] attributes = entry.getExtraAttributes();
-            for (int i = 0; i < attributes.length; i++) {
-                final IClasspathAttribute attribute = attributes[i];
-                final String name = attribute.getName();
-                if (name.equals("org.eclipse.jst.component.dependency")) {
-                    IPath location = getEntryLocation(entry, projPath);
-                    if (location != null) {
-                        if (location.toFile().getAbsolutePath().equals(file.getAbsolutePath()))
-                            return true;
-                    }
-                }
-            }
-        }
-
-        return false;
-    }
-
     private static boolean canAddReferencedComponentClasspathDependencies(IModule module, IProject refProject) {
-        if ("jst.web".equals(module.getModuleType().getId())) {
+        String type = module.getModuleType().getId();
+        if ("jst.web".equals(type) || "jst.ear".equals(type)) {
```

Comment:
Are we adding EAR support for shared library as part of this item? It looks like we were not allowing that before we make this change?

Reply:
It has always been supported for ear projects.  This code was added to fix a particular customer problem with a web project and I think whoever fixed that problem did not consider the ear case.  I was wondering if others should be added as well.  Looking at the plugin.xml for these properties it looks like they are shown for jst.web, jst.ear and jst.ejb.  So if anything I should be including jst.ejb as well:

   <page
   		id="com.ibm.ws.st.jee.ui.sharedLibraryWebProperties"
   		name="%sharedLibraryRefPropertyPage"
   		class="com.ibm.ws.st.jee.ui.internal.SharedLibPropertiesPage"
   		category="com.ibm.ws.st.ui.classpath">
   		<enabledWhen>
   		  <and>
   		   <adapt type="org.eclipse.core.resources.IProject">
   		   <test forcePluginActivation="false" property="com.ibm.ws.st.runtimeTarget"/>
   		   <or>
            <test forcePluginActivation="true"
              property="org.eclipse.wst.common.project.facet.core.projectFacet"
                value="jst.web"/>
            <test forcePluginActivation="true"
              property="org.eclipse.wst.common.project.facet.core.projectFacet"
                value="jst.ear"/>
            <test forcePluginActivation="true"
              property="org.eclipse.wst.common.project.facet.core.projectFacet"
                value="jst.ejb"/>
           </or>
           </adapt>
          </and>
   		</enabledWhen>
   	</page>
   </extension>

Reply:
sounds good.